libmate-panel-applet/mate-panel-applet.c

```diff
@@ -1431,15 +1512,18 @@ static void
 mate_panel_applet_set_background_string (MatePanelApplet *applet,
 				    const gchar *background)
 {
-	if (applet->priv->background == background)
+	MatePanelAppletPrivate *priv;
+
+	priv = mate_panel_applet_get_instance_private (applet);
+
+	if (priv->background == background)
 		return;
 
-	if (g_strcmp0 (applet->priv->background, background) == 0)
+	if (g_strcmp0 (priv->background, background) == 0)
 		return;
 
-	if (applet->priv->background)
-		g_free (applet->priv->background);
-	applet->priv->background = background ? g_strdup (background) : NULL;
```

Comment:
Is `if (priv->background)` line not missing here?

Reply:
There is no need to check if it is null before calling g_free:
```C
/**
 * g_free:
 * @mem: (nullable): the memory to free
 * 
 * Frees the memory pointed to by @mem.
 *
 * If @mem is %NULL it simply returns, so there is no need to check @mem
 * against %NULL before calling this function.
 */
void
g_free (gpointer mem)
{
  free (mem);
  TRACE(GLIB_MEM_FREE((void*) mem));
}
```